lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -194,55 +212,59 @@ def verify(payment_method, options = {})
 
       private
 
+      def off_session_request?(options = {})
+        (options[:off_session] || options[:setup_future_usage]) && options[:confirm] == true
+      end
+
       def add_connected_account(post, options = {})
         super(post, options)
         post[:application_fee_amount] = options[:application_fee] if options[:application_fee]
       end
 
       def add_whitelisted_attribute(post, options, attribute)
         post[attribute] = options[attribute] if options[attribute]
-        post
       end
 
       def add_capture_method(post, options)
         capture_method = options[:capture_method].to_s
         post[:capture_method] = capture_method if ALLOWED_METHOD_STATES.include?(capture_method)
-        post
       end
 
       def add_confirmation_method(post, options)
         confirmation_method = options[:confirmation_method].to_s
         post[:confirmation_method] = confirmation_method if ALLOWED_METHOD_STATES.include?(confirmation_method)
-        post
       end
 
       def add_customer(post, options)
         customer = options[:customer].to_s
         post[:customer] = customer if customer.start_with?('cus_')
-        post
       end
 
       def add_return_url(post, options)
         return unless options[:confirm]
 
         post[:confirm] = options[:confirm]
         post[:return_url] = options[:return_url] if options[:return_url]
-        post
       end
 
       def add_payment_method_token(post, payment_method, options)
         return if payment_method.nil?
 
         if payment_method.is_a?(ActiveMerchant::Billing::CreditCard)
-          p = create_payment_method(payment_method, options)
-          return p unless p.success?
+          if off_session_request?(options)
+            post[:payment_method_data] = create_payment_method_data(payment_method, options)
```

Comment:
What do you think about adding a `return` statement after this line? If I understand whats going on here, the `case` statement on 264 will not do anything more to this request. This is a fairly complex function so knowing that it is done here would be helpful.

Reply:
I agree, I mostly kept the logic the same (except replacing the original if/else for a case/when set).